account_move_accrual_entry/models/account_move.py

```diff
@@ -0,0 +1,278 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    accrual_date = fields.Date(string="Accrual Date")
+    accrual_account_move_ids = fields.Many2many(
+        comodel_name="account.move",
+        relation="accrual_account_move_rel",
+        column1="move_id",
+        column2="accrual_account_move_ids",
+        copy=False,
+        readonly=True,
+    )
+
+    def _get_accrual_account_id(self):
+        if self.accrual_date:
+            if not self.env.company.accrual_account_id:
+                raise UserError(
+                    _("Please set the accrual account in the invoincing settings.")
+                )
+        return self.env.company.accrual_account_id
+
+    def _get_accrual_asset_account_type_id(self):
+        if not self.env.company.accrual_asset_account_type_id:
+            raise UserError(
+                _("Please set the account type for assets in the invoincing settings.")
+            )
+        return self.env.company.accrual_asset_account_type_id
+
+    def _not_asset(self, x):
+        asset = self.env.company.accrual_asset_account_type_id.name
+        return x.accrual_account_id.user_type_id.name != asset
+
+    def _asset(self, x):
+        asset = self.env.company.accrual_asset_account_type_id.name
+        return x.accrual_account_id.user_type_id.name == asset
+
+    def _accrual_force_reconcile(self, rec):
+        invoice_line_ids = self.invoice_line_ids.filtered(
+            lambda x: x.accrual_account_id.user_type_id
+            != self._get_accrual_asset_account_type_id()
+        )
+        for l0, l1 in zip(invoice_line_ids, rec.line_ids):
+            (l0 | l1).reconcile()
+
+    def _create_line_ids(self):
+        line_ids = []
+        non_asset_invoice_lines = self.invoice_line_ids.filtered(
+            lambda x: x.accrual_account_id.user_type_id
+            != self._get_accrual_asset_account_type_id()
+        )
+        for idx in range(2):
+            for line in non_asset_invoice_lines:
+                ji = {
+                    "name": line.name,
+                    "date": self.accrual_date,
+                    "account_id": line.account_id.id
+                    if idx == 0
+                    else line.accrual_account_id.id,
+                    "partner_id": self.partner_id.id,
+                    "tax_tag_ids": line.tax_tag_ids.ids,
+                    "debit": line.price_subtotal if idx == 0 else 0,
+                    "credit": 0 if idx == 0 else line.price_subtotal,
+                }
+                line_ids.append((0, 0, ji))
```

Comment:
This should be moved to the create method, this functions should return only a list of dictionaries